controller/src/main/java/io/pravega/controller/server/bucket/BucketManager.java

```diff
@@ -105,6 +145,8 @@ CompletableFuture<Void> tryTakeOwnership(int bucket) {
                         
                         return bucketFuture;
                     } else {
+                        log.warn("{}: Unable to take ownership of bucket {} because another instance doesn't release it.",
```

Comment:
> What happens in this case? Is this controller going to retry taking ownership of this bucket or it will stop trying?

Bucket manager will retry (10 times) to take the ownership of bucket. 

> In that case, could the bucket be left orphan?

No, once the another controller successfully release the ownership, new owner will get a notification and it will start the bucket service. [Here](https://github.com/pravega/pravega/blob/dab2e15f251673264e92c6f004a2e7fa91ca51cb/controller/src/main/java/io/pravega/controller/server/bucket/ZooKeeperBucketManager.java#L79) is the implementation of it.